bugs/rhbz-1772505.txt

```diff
@@ -0,0 +1,274 @@
+https://bugzilla.redhat.com/show_bug.cgi?id=1772505
+
+=== Begin cloud-init SRU Template ===
+[Impact]
+
+Configuring cloud-init's swap cloud-config and setting the 'size' key to
+'auto' causes and error and prevents a swap file from being created and
+enabled.  This issue is only found on the Ubuntu Focal release.
+
+[Test Case]
+#!/bin/bash
+# Create a swap device with size='auto' and confirm swap file is created
+# Notes:
+#  1. using VMs as many LXD deployments use ZFS on which swapfiles do not
+#     work.
+#  2. This bug only affects Ubuntu Focal, the commit which introduced the
+#     regression (the fix for fallocate, on Jan 23; 6603706e).  19.4-33 is
+#     not affect, therefore we're only validating Focal release.
+#
+read -r -d '' CLOUDCFG <<- EOM
+#cloud-config
+swap:
+  filename: /test-swapfile
+  size: auto
+  maxsize: 2G
+EOM
+SETUP_PROPOSED=$(mktemp ubuntu-sru.XXX)
+cat > ${SETUP_PROPOSED} <<EOF
+#/bin/bash
+mirror=http://archive.ubuntu.com/ubuntu
+echo deb \$mirror \$(lsb_release -sc)-proposed main | tee /etc/apt/sources.list.d/proposed.list
+apt-get update -q
+apt-get install -qy cloud-init
+EOF
+BUGNUM="rhbz-1772505"
+VM_SLEEP=30
+
+cleanup() {
+    if [ -n "${SETUP_PROPOSED}" -a -e "${SETUP_PROPOSED}" ]; then
+        rm -f "${SETUP_PROPOSED}";
+    fi
+    lxc delete --force $name &>/dev/null
+}
+trap cleanup EXIT
+
+recreate() {
+    release=$1
+    name=${release}-recreate-${BUGNUM}
+
+    # initialize the contaier
+    lxc init images:ubuntu/$release/cloud --vm $name
```

Comment:
did you have to specify a vm profile (or do you have that as default in your lxc/lxd environment?

Reply:
oops n/m I see that you set the config below